src/libponyc/codegen/genreference.c

```diff
@@ -254,6 +254,7 @@ LLVMValueRef gen_localdecl(compile_t* c, ast_t* ast)
   // Store the alloca to use when we reference this local.
   codegen_setlocal(c, name, alloc);
 
+  /*
```

Comment:
Commenting out the debug info in gen_localdecl removes the boom from the 4369 example

Reply:
Would be an interesting next step to try to pare down to remove debug info for specific locals by name and see which specific local variable is breaking it.

Reply:
So, modify the generated LLVM ir for the given program?

Reply:
Joe was thinking "check the name and decide if to put" as using command line tools to link might change the situation.